### PR TITLE
refactor: move lines around, use const/let

### PR DESCRIPTION
**Pull request by staltz**
These are simple refactors, mostly just for my sanity as I try to navigate the codebase, hope it's useful:

- `require`s always at the top, unless needed for browser differentiation
- `const` preferred if we know we won't re-assign the thing afterwards
- `let` preferred over `var` (var allows you to do the declaration twice, allowing some sneaky bugs pass through sometimes)
- variables used in scope declared at the top, inner functions declared below that (see `db.js` for an example)

This PR doesn't contain anything else interesting other than code shuffling.

### CODE REVIEW

File: db.js [hunk 1]
```diff
@@ -226,20 +227,16 @@ exports.init = function (dir, config) {
...
 
   function clearIndexes() {
-    for (var index in indexes)
-      indexes[index].remove(() => {})
-  }
-
-  var indexes = {
-    base: baseIndex
+    for (const indexName in indexes)
+      indexes[indexName].remove(() => {})
```
Comment (staltz) [inline db.js:239]: The diff above looks weird, here's what's actually going on:

- `var indexes = {....` moved to the top of this file
- `for (var index....` changed to `for (const indexName....`

**Comment by arj03** [comment]
Was review this. Looks good except there are now merge conflicts. So once resolved, ok to merge.

**Comment by staltz** [comment]
Conflicts resolved :heavy_check_mark:

**Comment by arj03** [comment]
Nice

**Pull request merged**